src/log.h

```diff
@@ -32,21 +32,71 @@ class StreamProxy {
 public:
 	StreamProxy(std::ostream *os) : m_os(os) { }
 
+	static void fix_stream_state(std::ostream &os)
+	{
+		std::ios::iostate state = os.rdstate();
+		// clear error state so the stream works again
+		os.clear();
+		if (state & std::ios::eofbit)
+			os << "(ostream:eofbit)";
+		if (state & std::ios::badbit)
+			os << "(ostream:badbit)";
+		if (state & std::ios::failbit)
+			os << "(ostream:failbit)";
+	}
```

Comment:
if this is static with an implementation shouldn't it go to log.cpp?
AFAIK compilers will possibly decide to inline this anyway.

Reply:
Yea... although with LTO, can't it be inlined anyway?

I would much prefer to use [[likely]] and [[unlikely]], but unfortunately that's C++20.

Reply:
> Although with LTO, can't it be inlined anyway?

True.